bin/notify-on-record-change

```diff
@@ -13,8 +13,20 @@ source_name=${3:?A record source name is required as the third argument.}
 # if the file is not already present, just exit
 "$bin"/s3-object-exists "$dst" || exit 0
 
+s3path="${dst#s3://}"
+bucket="${s3path%%/*}"
+key="${s3path#*/}"
+
 src_record_count="$(wc -l < "$src")"
-dst_record_count="$(wc -l < <(aws s3 cp --no-progress "$dst" - | xz -T0 -dcfq))"
+
+# Try getting record count from S3 object metadata
+dst_record_count="$(aws s3api head-object --bucket "$bucket" --key "$key" --query Metadata.recordcount --output text 2>/dev/null | tr -d '"')"
```

Comment:
With `--output text` it shouldn't be necessary to remove quotes with `tr` (unless extra quotes were added at some point by bad quoting, but quoting during upload looks fine now).